
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Assembly analysis &#8212; Biata-2021 0.0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="assembly-analysis">
<h1>Assembly analysis<a class="headerlink" href="#assembly-analysis" title="Permalink to this headline">¶</a></h1>
<p>In the presentation we will cover:</p>
<ul class="simple">
<li><p>Overview of MGnify annotation pipeline</p></li>
<li><p>Taxonomic assignment</p></li>
<li><p>Functional characterisation</p></li>
<li><p>Pathways/systems</p></li>
<li><p>Using the contig viewer</p></li>
</ul>
<section id="mgnify-hands-on-exercises">
<h2>MGnify hands-on exercises<a class="headerlink" href="#mgnify-hands-on-exercises" title="Permalink to this headline">¶</a></h2>
<p>For this session we will look at some of the data and analyses that are available from MGnify. We will navigate the resource, try out different ways to search for interesting samples/studies, and then investigate the analysis results that are available for assemblies.</p>
<section id="browsing-mgnify">
<h3>Browsing MGnify<a class="headerlink" href="#browsing-mgnify" title="Permalink to this headline">¶</a></h3>
<p>From the MGnify front page (<a class="reference external" href="https://www.ebi.ac.uk/metagenomics/">https://www.ebi.ac.uk/metagenomics/</a>) you can see various options to browse the data. There are quick links to the various data-types (e.g. amplicon, assembly, metagenomes, etc) we support, as well as a subset of the biomes that the data covers.</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Click on the “wastewater” biome icon.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many studies does MGnify hold that relate to wastewater?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many samples does that relate to?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> From the sample page, filter the rows with accession ERS1215575, and take a look at the metadata available.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> Do you know the exact location of where the sample was taken?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> What are the lat/long co-ordinates?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> Follow the link to the BioSamples record, can you find any more information about the location of the sample?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> From the tabs in the header bar, select <strong>Text search</strong>, and then select <strong>Samples</strong> below the search box. There are a number of metadata fields available to allow you to filter for a sample of interest to you. Not all are relevant to all samples.
Within the hierarchy of biomes, navigate to environmental&gt;aquatic&gt;lentic. You should see 92 samples. Now select the depth filter.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many lentic samples have depth data associated with them?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> Using the sliders, can you identify a sample of a lentic water system from a depth between 47-49m?</p>
</section>
<section id="mgnify-assembly-analysis">
<h3>MGnify assembly analysis<a class="headerlink" href="#mgnify-assembly-analysis" title="Permalink to this headline">¶</a></h3>
<p>Now we will look at some assembly data that has been analysed by MGnify.</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Search for <strong>MGYS00003598</strong>, and go to this study page. This is a large study where MGnify have assembled the raw reads from an existing public study. The list of assemblies is shown at the bottom of the study page.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many analyses are included in this study?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Click on the 2nd analysis link in the list <strong>MGYA00510849</strong>. You could alternatively search for this accession using the text search options. Have a look at the information within the <strong>Quality control</strong> tab.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many contigs are included in this analysis?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> What length is the longest contig in this dataset?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Click on the <strong>Taxonomic analysis</strong> tab and examine the phylum composition in the graphs and the krona plot.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> What proportion of the total LSU rRNA predictions are eukaryotic?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Try switching between the other available graph views.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> Which phylum contains the highest proportion of LSU rRNA predictions?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Click on the <strong>Functional analysis</strong> tab. The top part of this page shows a sequence feature summary, showing the number of contigs with predicted coding sequences (pCDS), the number of pCDS with InterPro matches, etc.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many predicted coding sequences (pCDS) are in the assembly?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many pCDS have InterProScan hits?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Scroll down the page to the InterPro match summary section.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many different InterPro entries are matched by the pCDS?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> Why is this number different to the number of pCDS that have InterProScan hits?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Click on the <strong>GO Terms</strong> sub-tab. This shows a summary of the most common GO terms annotated to the pCDS as both bar charts, and pie charts.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> What are the top 3 biological process terms predicted for the pCDS from this assembly?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Have a look at the information in the <strong>Pfam</strong> and <strong>KO</strong> (KEGG orthologue) sub-tabs.</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Click on the <strong>Pathways/Systems</strong> tab. Have a look at the data reported in the 3 sub-tabs: KEGG Module, Genome Properties, and antiSMASH.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many KEGG modules are reported for this assembly?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many of these are 100% complete (i.e. all of the constituent KOs are found)?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many Genome Properties of the category <strong>DNA handling</strong>, are found within this assembly?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> What is the most common class of biosynthetic gene cluster found in this assembly?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How many non-ribosomal peptide synthetase gene clusters are identified by antiSMASH in this assembly?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Click on the <strong>Contig Viewer</strong> tab. Load the data for the 4th contig in the list by clicking on the contig name (ERZ501066.4-NODE-4-length-276957-cov-33.799655). This contig will now be loaded into the viewer.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> How long is this contig?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> The longest pCDS in the contig appears to start at 202339. What protein is coded for?</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> Looking at the antiSMASH annotations, where within the contig do any transport-related genes fall?</p>
<p><a class="reference internal" href="_images/action.png"><img alt="action" src="_images/action.png" style="width: 0.25in; height: 0.25in;" /></a> Zoom into that region to see the predicted regions in more detail. Have a look at the information about the various transport-related genes.</p>
<p><a class="reference internal" href="_images/question.png"><img alt="question" src="_images/question.png" style="width: 0.26667in; height: 0.26667in;" /></a> What region of the contig is predicted to code for a major facilitator transporter?</p>
<p><a class="reference internal" href="_images/info.png"><img alt="info" src="_images/info.png" style="width: 0.26667in; height: 0.26667in;" /></a> There are lots of different visualisation options available within the contig viewer. Take some time now to investigate the various options, and play about with it by looking at a few different contigs and the anotations they contain.</p>
</section>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="docs.html">Biata-2021</a></h1>








<h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="docs.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2021, MGnify.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 4.0.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/assembly-2.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>